tests/baselines/reference/errorOnUnionVsObjectShouldDeeplyDisambiguate.errors.txt

```diff
@@ -37,41 +37,51 @@ tests/cases/compiler/errorOnUnionVsObjectShouldDeeplyDisambiguate.ts(27,16): err
                    ~~~~~~~
 !!! error TS2322: Type 'string' is not assignable to type 'Promise<string>'.
 !!! related TS6502 tests/cases/compiler/errorOnUnionVsObjectShouldDeeplyDisambiguate.ts:3:8: The expected type comes from the return type of this signature.
+!!! related TS1356 tests/cases/compiler/errorOnUnionVsObjectShouldDeeplyDisambiguate.ts:18:10: Did you mean to mark this function as 'async'?
           c: () => "hello",
                    ~~~~~~~
 !!! error TS2322: Type 'string' is not assignable to type 'Promise<string>'.
 !!! related TS6502 tests/cases/compiler/errorOnUnionVsObjectShouldDeeplyDisambiguate.ts:4:8: The expected type comes from the return type of this signature.
+!!! related TS1356 tests/cases/compiler/errorOnUnionVsObjectShouldDeeplyDisambiguate.ts:19:10: Did you mean to mark this function as 'async'?

```

Comment:
This is kind of a weird thing now that I think about it - why would I want to mark this as async? There's no `await` anywhere there.

If a user *did* write `await` in the function body, we already provide the same error message/quick fix to mark the function as `async`, so this could be annoying.

```ts
interface Stuff {
    b: () => Promise<string>;
}

function foo(): Stuff | Date {
    return {
        b: _ =>
          await "hello" // two error spans with duplicate related spans
        
    }
}
```

So this is only distinct actionable advice information for a user if they *hadn't* written `await`. Would making this `async` actually be the right thing? It *does* come up, it's unclear if it's the real fix to a problem.

Reply:
> This is kind of a weird thing now that I think about it - why would I want to mark this as async? There's no await anywhere there.

Because `async () => "constant"` is way, way more canonical and concise than `() => Promise.resolve("constant")` when the API you're using expects a function returning a promise.

> So this is only distinct actionable advice information for a user if they hadn't written await

More like, using `await` means the containing function should be async to make what you're trying to do work, while if assignability to a promise return fails, it should be async to satisfy the API. Different reasons entirely that happen to have the same fix.